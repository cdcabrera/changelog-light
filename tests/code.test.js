const util = require('util');
const { exec } = require('child_process');
const promiseExec = util.promisify(exec);

describe('General code checks', () => {
  const srcDir = 'src';

  it('should only have specific console.[warn|log|info|error] methods', async () => {
    const { stdout } = await promiseExec(
      `echo "$(cd ./${srcDir} && git grep -n -E "(console.warn|console.log|console.info|console.error)")"`
    );

    expect(
      stdout
        .toString()
        .trim()
        .split(/[;]/g)
        .map(value => value.replace(/[\n\r]/g, ' ').trim())
        .filter(value => value.length)
    ).toMatchSnapshot('console methods');
  });
});
