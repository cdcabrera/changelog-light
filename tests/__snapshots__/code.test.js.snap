// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`General code checks should only have specific console.[warn|log|info|error] methods: console methods 1`] = `
[
  "cmds.js:21:    console.error(color.RED, errorMessage.replace('{0}', e.message), color.NOCOLOR)",
  "cmds.js:138:    console.error(color.RED, \`Semver: \${e.message}\`, color.NOCOLOR)",
  "cmds.js:168:    console.error(color.RED, \`Semver: \${e.message}\`, color.NOCOLOR)",
  "files.js:69:    console.info(\` \${updatedBody}\`)",
  "global.js:64:    console.error(color.RED, \`Conventional commit types: \${e.message}\`, color.NOCOLOR)",
  "index.js:42:    console.info( index.js:56:    console.info(color.NOCOLOR)",
]
`;
